### PR TITLE
feat: add isValidGermanDate and isValidTime

### PR DESCRIPTION
**Pull request by Ralf77**

### CODE REVIEW

File: src/date.test.ts [hunk 1]
```diff
@@ -63,6 +65,36 @@ describe('formatGermanDateTime', () => {
   });
 });
 
+describe('isValidGermanDate', () => {
+  it('should determine if a string matches the german date format dd.mm.yyyy', () => {
```
Comment (spawnia) [inline src/date.test.ts:69]:
```suggestion
  it('determines if a string is a german date, day or month being one or two digits and year four digits, separated by dots', () => {
```

File: src/date.test.ts [hunk 2]
```diff
@@ -63,6 +65,36 @@ describe('formatGermanDateTime', () => {
...
+    expect(isValidGermanDate('01.yy.2000')).toEqual(false);
+    expect(isValidGermanDate('01.01.zzzz')).toEqual(false);
+    expect(isValidGermanDate('')).toEqual(false);
+    expect(isValidGermanDate('..')).toEqual(false);
+    expect(isValidGermanDate('abc')).toEqual(false);
+  });
+});
+
+describe('isValidTime', () => {
+  it('should determine if a string matches the time format HH:mm', () => {
```
Comment (spawnia) [inline src/date.test.ts:87]:
```suggestion
  it('determines if a string matches the time format HH:mm', () => {
```

File: src/date.ts [hunk 3]
```diff
@@ -15,6 +15,26 @@ export function isFuture(date: Date | string | number): boolean {
...
 }
 
+export function isValidGermanDate(date: string): boolean {
+  const regexGermanDateFormat = /^(\d{1,2}[.]\d{1,2}[.]\d{4})$/;
+
+  if (date.match(regexGermanDateFormat)) {
+    return true;
+  }
+
+  return false;
```
Comment (spawnia) [inline src/date.ts:25]:
```suggestion
return /^(\d{1,2}[.]\d{1,2}[.]\d{4})$/.test(date);
```

File: src/date.ts [hunk 4]
```diff
@@ -15,6 +15,26 @@ export function isFuture(date: Date | string | number): boolean {
...
+}
+
+export function isValidTime(time: string): boolean {
+  const regexTimeFormat = /^(\d{2}:\d{2})$/;
+
+  if (time.match(regexTimeFormat)) {
+    return true;
+  }
+
+  return false;
```
Comment (spawnia) [inline src/date.ts:35]:
```suggestion
  return /^(\d{2}:\d{2})$/.test(time);
```

**Review by spawnia — CHANGES_REQUESTED** [review]

File: src/date.ts [hunk 5]
```diff
@@ -15,6 +15,14 @@ export function isFuture(date: Date | string | number): boolean {
   return isAfter(date, new Date());
 }
 
+export function isValidGermanDate(date: string): boolean {
+  return /^(\d{1,2}[.]\d{1,2}[.]\d{4})$/.test(date);
+}
+
+export function isValidTime(time: string): boolean {
```
Comment (spawnia) [inline src/date.ts:22]: Actually I think the name for this is not specific enough. How about `isHoursAndMinutes`?

Comment (Ralf77) [reply src/date.ts:22]: How about "isTimeWithHoursAndMinutes"?

Comment (spawnia) [reply src/date.ts:22]: Sounds good.

**Pull request merged**